src/main/java/com/github/princesslana/eriscasper/ErisCasper.java

```
@@ -60,13 +60,21 @@ private ErisCasper(BotToken token, Optional<ShardPayload> shard) {
   }
 
   public void run(Bot bot) {
-    Observable<Event> events = getEvents();
-
-    RepositoryManager rm = RepositoryManager.create(events);
-
-    bot.apply(new BotContext(events, routes, rm))
-        .doOnError(t -> LOG.warn("Exception thrown by Bot", t))
-        .blockingAwait();
+    try {
+      getEvents()
+          .compose(
+              evts ->
+                  bot.apply(new BotContext(evts, routes, RepositoryManager.create(evts)))
+                      .toObservable())
+          .doOnError(t -> LOG.warn("Exception thrown by Bot", t))
+          .blockingSubscribe();
+    } finally {
```

Should we catch and display `ErisCasperFatalException`s here?

---

They would be displayed in the `doOnError`, or we just let them bubble up, where if the bot dev isn't catching them they'll display when the program exits anyway.